tools/accuracy_checker/openvino/tools/accuracy_checker/adapters/palm_detection.py

```diff
@@ -0,0 +1,255 @@
+                                                                       min_scale) * stride_index / (num_strides - 1)
+
+    def generate_anchors(self):
+        anchors = []
+        layer_id = 0
+        while layer_id < self.num_layers:
+            anchor_height = []
+            anchor_width = []
+            aspect_ratios = []
+            scales = []
+
+            last_same_stride_layer = layer_id
+            while last_same_stride_layer < len(self.strides) and (self.strides[last_same_stride_layer] ==
+                                                                  self.strides[layer_id]):
+                scale = self.calculate_scale(self.min_scale, self.max_scale, last_same_stride_layer, len(self.strides))
+                ar_and_s = zip([1, 2, 0.5], [0.1, scale, scale]) if (
+                    last_same_stride_layer == 0) and self.reduce_boxes_in_lowest_layer else zip(
+                    self.aspect_ratios, [scale] * len(self.aspect_ratios))
+                for aspect_ratio, scale_ in ar_and_s:
+                    aspect_ratios.append(aspect_ratio)
+                    scales.append(scale_)
+
+                if self.inteprolated_scale_aspect_ratio > 0:
+                    scale_next = 1 if last_same_stride_layer == len(self.strides) - 1 else self.calculate_scale(
+                        self.min_scale, self.max_scale, last_same_stride_layer + 1, len(self.strides))
+                    scales.append(np.sqrt(scale * scale_next))
+                    aspect_ratios.append(self.inteprolated_scale_aspect_ratio)
+                last_same_stride_layer += 1
+
+            for aspect_ratio, scale in zip(aspect_ratios, scales):
+                anchor_height.append(scale / np.sqrt(aspect_ratio))
+                anchor_width.append(scale * np.sqrt(aspect_ratio))
+
+            feature_map_height = self.feature_map_height[layer_id] if self.feature_map_height else int(
+                np.ceil(self.input_size_height / self.strides[layer_id]))
+            feature_map_width = self.feature_map_width[layer_id] if self.feature_map_height else int(
+                np.ceil(self.input_size_width / self.strides[layer_id]))
+
+            for y in range(feature_map_height):
+                for x in range(feature_map_width):
+                    for anchor_w, anchor_h in zip(anchor_width, anchor_height):
+                        anchor = [(x + self.anchor_offset_x) / feature_map_width,
+                                  (y + self.anchor_offset_y) / feature_map_height,
+                                  1 if self.fixed_anchor_size else anchor_w,
+                                  1 if self.fixed_anchor_size else anchor_h]
+
+                        anchors.append(anchor)
+
+            layer_id = last_same_stride_layer
+
+        return np.array(anchors)
+
+    def decode_boxes(self, raw_boxes):
+        boxes = []
+        num_boxes, _ = raw_boxes.shape
+
+        for i in range(num_boxes):
```

Comment:
as I can see there is no any changes in number of boxes, can we vectorize this cycle? just operate with numpy array instead?

Reply:
fixed